conan/conanfile.py

```diff
@@ -0,0 +1,61 @@
+###############################################################################
+##  This file is part of the KDAB State Machine Editor Library.
+##
+##  Copyright (C) 2014-2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com.
+##  All rights reserved.
+##
+##  Licensees holding valid commercial KDAB State Machine Editor Library
+##  licenses may use this file in accordance with the KDAB State Machine Editor
+##  Library License Agreement provided with the Software.
+##
+##  This file may be distributed and/or modified under the terms of the
+##  GNU Lesser General Public License version 2.1 as published by the
+##  Free Software Foundation and appearing in the file LICENSE.LGPL.txt included.
+##
+##  This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+##  WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+##
+##  Contact info@kdab.com if any conditions of this licensing are not
+##  clear to you.
+###############################################################################
+
+from conans import ConanFile, CMake, tools
+
+
+class KDStateMachineEditorConan(ConanFile):
+    name = "KDStateMachineEditor"
+    version = "1.2"
+    license = "https://raw.githubusercontent.com/KDAB/KDStateMachineEditor/1.2/LICENSE.txt"
+    author = "Renato Araujo Oliveira Filho <renato.araujo@kdab.com>"
+    url = "https://github.com/KDAB/KDStateMachineEditor.git"
+    description = "The KDAB State Machine Editor Library is a framework that can be used to help develop full-featured State Machine Editing graphical user interfaces and tools."
+    generators = "cmake"
+
+    def requirements(self):
+        self.requires("qt/5.13.2@kdab/stable")
+        self.requires("KDChart/2.6.50@kdab/stable")
+
+    def source(self):
+        git = tools.Git(folder="")
+        git.clone(self.url)
+        git.checkout("%s"%self.version)
+
+    def configure(self):
+        self.options["qt"].qtsvg = True
+        self.options["qt"].qtdeclarative = True
+        self.options["qt"].qtremoteobjects = True
+        self.options["qt"].qtscxml = True
+
+    def build(self):
+        self.cmake = CMake(self)
+        self.cmake.definitions["WITH_INTERNAL_GRAPHVIZ"] = True
+        self.cmake.definitions["BUILD_EXAMPLES"] = False
+        self.cmake.definitions["BUILD_DOCS"] = False
```

Comment:
can we pass the cmake options to the conan install command for installing the docs or examples?
someone may want to build the docs or examples?

Also I see that we should have a cmake option to disable building the tests. I 'll add that to the kdstatemachineeditor/CMakeLists.txt

Reply:
I just added a new cmake option 'BUILD_TESTS' to the 1.2 and master branches.
Please add support for BUILD_TESTS.